Libraries/NavigationExperimental/NavigationTypeDefinition.js

```diff
@@ -111,4 +113,25 @@ export type NavigationStyleInterpolator = (
   props: NavigationSceneRendererProps,
 ) => Object;
 
-export type NavigationTransitionConfigurator = () => NavigationTransitionSpec;
+export type NavigationTransitionConfigurator = (
+  transitionProps: NavigationTransitionProps,
+  prevTransitionProps: ?NavigationTransitionProps,
+) => NavigationTransitionSpec;
+
+export type AnimationConfig = Object;
+
+export type AnimationEndResult = {
+  finished: bool
+};
+
+export type AnimationEndCallback = (result: AnimationEndResult) => void;
+
+export type CompositeAnimation = {
+  start: (callback?: ?AnimationEndCallback) => void;
+  stop: () => void;
+};
+
+export type NavigationTransitionTiming = (
+  value: NavigationAnimatedValue,
+  config: AnimationConfig,
+) => CompositeAnimation;
```

Comment:
I totally agree. 👍 Updated the PR. 

Should we remove `NavigationAnimatedValue` as well? Perhaps the best way to approach this is to expose the type definitions from `Animated` itself so we can just import them from there. I might submit another PR to fix this in `Animated`.

🍺